egs/gigaspeech/s5/run.sh

```diff
@@ -1,143 +1,133 @@
 #!/usr/bin/env bash
-# Copyright 2021 Xiaomi Corporation (Author: Yongqing Wang)
+# Copyright 2021  Xiaomi Corporation (Author: Yongqing Wang)
+#                 Seasalt AI, Inc (Author: Guoguo Chen)
 # Apache 2.0
 
+set -e
+set -o pipefail
+
 . ./cmd.sh
 . ./path.sh
 
 stage=0
 
-gigaspeech_root=~/GigaSpeech_data/
-train_sets=train
-test_sets="dev test"
-lm_dir=data/local/lm
-dict_dir=data/local/dict
+# GigaSpeech configurations.
+gigaspeech_root=/data/GigaSpeech_Data/
+gigaspeech_train_subset=XL
+gigaspeech_test_sets="gigaspeech_dev gigaspeech_test"
+gigaspeech_train_sets="gigaspeech_train_${gigaspeech_train_subset,,}"
 
-. ./cmd.sh
-. ./path.sh
-. parse_options.sh
+# Train/Dev/Test sets.
+test_sets="$gigaspeech_test_sets"
+train_sets="$gigaspeech_train_sets"
+train_combined="$gigaspeech_train_sets"
 
-# you might not want to do this for interactive shells.
-set -e
+# G2P models.
+g2p_model=$gigaspeech_root/dict/g2p/g2p.model.4
 
-for f in $lm_dir $dict_dir; do
-  [ ! -d $f ] && mkdir -p $f
-done
-g2p_model_dir=$gigaspeech_root/dict/g2p
-lexicon_gigaspeech=$gigaspeech_root/dict/lexicon_gigaspeech.txt
-order=4
+# Experiment configurations.
+train_nj=100
+decode_nj=100
+lm_order=4
+lm_dir=data/local/lm
+dict_dir=data/local/dict
 
 if [ $stage -le 0 ]; then
-  echo -e "======Download GigaSpeech START|current time : `date +%Y-%m-%d-%T`======"
-  # Download and analyze GigaSpeech datasets
-  # cd GigaSpeech to run utils/gigaspeech_download.sh to download data
-  git clone https://github.com/SpeechColab/GigaSpeech.git
-  pushd GigaSpeech
-  utils/gigaspeech_download.sh $gigaspeech_root
-  popd
-  echo -e "======Download GigaSpeech END|current time : `date +%Y-%m-%d-%T`======"
+  echo "======Prepare GigaSpeech START | current time : `date +%Y-%m-%d-%T`===="
+  local/gigaspeech_data_prep.sh \
+    --stage 0 \
+    --train-subset $gigaspeech_train_subset \
+    $gigaspeech_root data/ || exit 1;
+  echo "======Prepare GigaSpeech END | current time : `date +%Y-%m-%d-%T`======"
 fi
 
-# This stage maybe lead to errors,
-# because stage1 in gigaspeech_data_prep.sh should be run in path of gigaspeech
-# stage 3 in gigaspeech_data_prep.sh should be run here
 if [ $stage -le 1 ]; then
-  echo -e "======Prepare Data START|current time : `date +%Y-%m-%d-%T`======"
-  # Prepare GigaSpeech data
-  GigaSpeech/toolkits/kaldi/gigaspeech_data_prep.sh $gigaspeech_root data true
-  echo -e "======Prepare Data END|current time : `date +%Y-%m-%d-%T`======"
+  echo "======Prepare Dictionary START | current time : `date +%Y-%m-%d-%T`===="
+  [ ! -f $g2p_model ] && echo "$0: Cannot find G2P model $g2p_model" && exit 1
```

Comment:
we need to `mkdir -p $dict_dir/cmudict; cp $gigaspeech_root/dict/cmudict.0.7a $dict_dir/cmudict` before `local/prepare_dict.sh`, since `local/prepare_dict.sh` assumes cmudict is there

Reply:
local/prepare_dict.sh actually creates the folder, isn't it? https://github.com/SpeechColab/kaldi/pull/2/files#diff-1e43fbaf27de260b699e5fee0b4da8ab98d17dc430dd3c7cd2365d9024d1a9a3R42

Reply:
yeah, I thought `local/prepare_dict.sh` would download a new one, instead of copying the one we provided in Giga dataset. but actually they are the same.